### ISSUE TITLE
Move all our adapters to ListAdapter base class

### ISSUE THREAD

**Issue by semanticer** [toggl/mobileapp#5925]
## 🚧 Refactor Proposal

Related slack thread https://toggl.slack.com/archives/C02APMYTM/p1568810483151900

Our implementation of `BaseRecyclerAdapter` is complicated and there are now several issues that might be related to some unpredictable behaviors (#5692 #5691 ). Therefore, we want to move to standard `ListAdapter` [docs](https://developer.android.com/reference/android/support/v7/recyclerview/extensions/ListAdapter) and let google handle all of these issues.

## Dependencies

First, we need to verify that `ListAdapter` is available and works correctly on Xamarin.

## Technical aspects

Basically we need to replace `BaseRecyclerAdapter` everywhere as a base class and use the `ListAdapter` instead.  Extending the `ListAdapter` is pretty straight forward, just follow the [docs](https://developer.android.com/reference/android/support/v7/recyclerview/extensions/ListAdapter)

**Comment by heytherewill** [comment]
I'm un-assigning myself from this since I have other things to work on, but here [a WIP branch](https://github.com/toggl/mobileapp/tree/better-adapters) with the bulk of the needed changes.

We basically need to make it compiler and see if the fact that all items returned from the ListAdapter are now java objects will be a problem or not.

**Comment by semanticer** [comment]
After discussion here https://toggl.slack.com/archives/CD40QCV25/p1569248243020200 and failure of approach tried in #5940 we decided to move this issue to the next release (2.5) and we will probably try to rewrite some of the ListAdapter's code into C#

**Comment by semanticer** [comment]
Also un-assigning myself for now but I'm willing to try again!

**Comment by semanticer** [comment]
My latest attempt on this is still [WIP](https://github.com/toggl/mobileapp/tree/wip-list-adapter) https://github.com/toggl/mobileapp/tree/wip-list-adapter I'm essentially trying to rewrite ListAdapter and AsyncListDiffer to C# and incorporate our concepts of `DiffingStrategy` etc. There are still couple of issues and I'm still not 100% it will work out in the end. Therefore, I suggest removing this from 2.5. I would like to give this a couple more hours to decide if there is some light at the end of the tunnel in 2.6.

**Comment by semanticer** [comment]
Let's end this madness. I went pretty deep into reimplementing `ListAdapter` and `AsyncListDiffer` in C# and I've tried to use Google's building blocks as much as possible but the generics limitation forced me to go pretty deep and reimplement a lot of things anyway. The resulting implementation was very close to what we have right now (https://github.com/toggl/mobileapp/tree/wip-list-adapter) and I encountered weird cryptic crashes like: `A/libc: Fatal signal 11 (SIGSEGV), code 2 (SEGV_ACCERR), ` and I don't think it makes much sense to continue to try and fix this because there is no big gain at the end of the tunnel anyway.  

I consider Adapters and ListViews the most basic building blocks of any UI toolkit and the fact that there is no solution for them in Xamarin and nobody even cares makes me think that Xamarin is even more doomed than I originally thought.
